safe/gui/tools/wizard/step_kw40_classify.py

```diff
@@ -77,6 +78,26 @@ def get_next_step(self):
         else:
             subcategory = {'key': None}
 
+        layer_data_provider = self.parent.layer.dataProvider()
+        # Iterate through all inasafe fields
+        for inasafe_field in self.inasafe_fields_for_the_layer():
```

Comment:
Hi @myarjunar 
I think you just need to check whether `self.inasafe_fields_for_the_layer()` is empty or not.

And in this code, it will go directly to `self.parent.step_kw_source` if it doesn't have `inasafe_fields`. What if it has inasafe_default_fields? It should go to inasafe_default_fields step first not go directly to `step_kw_source`.